packages/crab-netting/src/CrabNetting.sol

```diff
@@ -1,55 +1,992 @@
+                toRemove -= r.amount;
+                delete deposits[userDepositsIndex[msg.sender][i - 1]];
+            }
+        }
+        IERC20(usdc).transfer(msg.sender, _amount);
+
+        emit USDCDeQueued(msg.sender, _amount, usdBalance[msg.sender]);
+    }
+
+    /**
+     * @notice queue Crab for withdraw from crab strategy
+     * @param _amount crab amount to withdraw
+     */
+    function queueCrabForWithdrawal(uint256 _amount) external {
+        require(
+            _amount >= minCrabAmount,
+            "withdraw amount smaller than minimum OTC amount"
+        );
+        IERC20(crab).transferFrom(msg.sender, address(this), _amount);
+        crabBalance[msg.sender] = crabBalance[msg.sender] + _amount;
+        withdraws.push(Receipt(msg.sender, _amount));
+        userWithdrawsIndex[msg.sender].push(withdraws.length - 1);
+        emit CrabQueued(
+            msg.sender,
+            _amount,
+            crabBalance[msg.sender],
+            withdraws.length - 1
+        );
+    }
+
+    /**
+     * @notice withdraw Crab from queue
+     * @param _amount Crab amount to dequeue
+     */
+    function withdrawCrab(uint256 _amount) external {
+        require(!isAuctionLive, "auction is live");
+        crabBalance[msg.sender] = crabBalance[msg.sender] - _amount;
+        require(
+            crabBalance[msg.sender] >= minCrabAmount ||
+                crabBalance[msg.sender] == 0,
+            "remaining amount smaller than minimum, consider removing full balance"
+        );
+        // remove that _amount the users last deposit
+        uint256 toRemove = _amount;
+        uint256 lastIndexP1 = userWithdrawsIndex[msg.sender].length;
+        for (uint256 i = lastIndexP1; i > 0; i--) {
+            Receipt storage r = withdraws[
+                userWithdrawsIndex[msg.sender][i - 1]
+            ];
+            if (r.amount > toRemove) {
+                r.amount -= toRemove;
+                toRemove = 0;
+                break;
+            } else {
+                toRemove -= r.amount;
+                delete withdraws[userWithdrawsIndex[msg.sender][i - 1]];
+            }
+        }
+        IERC20(crab).transfer(msg.sender, _amount);
+        emit CrabDeQueued(msg.sender, _amount, crabBalance[msg.sender]);
+    }
+
+    /**
+     * @dev swaps _quantity amount of usdc for crab at _price
+     * @param _price price of crab in usdc
+     * @param _quantity amount of USDC to net
+     */
+    function netAtPrice(uint256 _price, uint256 _quantity) external onlyOwner {
+        _checkCrabPrice(_price);
+        uint256 crabQuantity = (_quantity * 1e18) / _price;
+        require(
+            _quantity <= IERC20(usdc).balanceOf(address(this)),
+            "Not enough deposits to net"
+        );
+        require(
+            crabQuantity <= IERC20(crab).balanceOf(address(this)),
+            "Not enough withdrawals to net"
+        );
+
+        // process deposits and send crab
+        uint256 i = depositsIndex;
```

Comment:
Is depositsIndex initialized to zero before netAtPrice or depositAuction is called?

Reply:
depositsIndex is a storage variable. It stores the index of the last processed Deposit Receipt